src/pages/workspaces/hooks/useCloudEnvironmentPolling.ts

```diff
@@ -0,0 +1,98 @@
+
+export interface CloudEnvironmentDetails {
+  runtimes?: ListRuntimeItem[];
+  refreshRuntimes: (maybeStale?: boolean) => Promise<void>;
+  persistentDisks?: PersistentDisk[];
+  appDataDisks?: PersistentDisk[];
+}
+
+export const useCloudEnvironmentPolling = (
+  name: string,
+  namespace: string,
+  workspace?: Workspace
+): CloudEnvironmentDetails => {
+  const [controller, setController] = useState(new window.AbortController());
+  const abort = () => {
+    controller.abort();
+    setController(new window.AbortController());
+  };
+  const signal = controller.signal;
+
+  const timeout = useRef<NodeJS.Timeout>();
+  const [runtimes, setRuntimes] = useState<ListRuntimeItem[]>();
+  const [persistentDisks, setPersistentDisks] = useState<PersistentDisk[]>();
+  const [appDataDisks, setAppDataDisks] = useState<PersistentDisk[]>();
+
+  const saturnWorkspaceNamespace = workspace?.workspace?.namespace;
+  const saturnWorkspaceName = workspace?.workspace?.name;
+
+  const reschedule = (ms) => {
+    clearTimeout(timeout.current);
+    timeout.current = setTimeout(refreshRuntimesSilently, ms);
+  };
+  const load = async (maybeStale?: boolean): Promise<void> => {
+    try {
+      const cloudEnvFilters = _.pickBy((l) => !_.isUndefined(l), {
+        role: 'creator',
+        saturnWorkspaceName,
+        saturnWorkspaceNamespace,
+      }) as Record<string, string>; // we literally just filtered out the undefined values, but ts doesn't know this
+
+      // Disks.list API takes includeLabels to specify which labels to return in the response
+      // Runtimes.listV2 API always returns all labels for a runtime
+      const [newDisks, newRuntimes] = await Promise.all([
+        Ajax(signal)
+          .Disks.disksV1()
+          .list({
+            ...cloudEnvFilters,
+            includeLabels: 'saturnApplication,saturnWorkspaceName,saturnWorkspaceNamespace',
+          }),
+        Ajax(signal).Runtimes.listV2(cloudEnvFilters),
+      ]);
+
+      setRuntimes(newRuntimes);
+      setAppDataDisks(_.remove((disk) => _.isUndefined(getDiskAppType(disk)), newDisks));
+      setPersistentDisks(_.filter((disk) => _.isUndefined(getDiskAppType(disk)), newDisks));
+      const runtime = getCurrentRuntime(newRuntimes);
+      reschedule(
+        maybeStale ||
+          ['Creating', 'Starting', 'Stopping', 'Updating', 'LeoReconfiguring'].includes(
+            getConvertedRuntimeStatus(runtime) ?? ''
+          )
+          ? 10000
+          : 120000
+      );
+    } catch (error) {
+      reschedule(30000);
+      throw error;
+    }
+  };
+  const refreshRuntimes = withErrorReporting('Error loading cloud environments', load) as (
+    maybeStale?: boolean
+  ) => Promise<void>;
+  const refreshRuntimesSilently = withErrorIgnoring(load);
+  useEffect(() => {
+    if (
+      workspace?.workspaceInitialized &&
+      workspace.workspace.name === name &&
```

Comment:
This prevents a double load by waiting for the correct workspace to be loaded for the page. 
Even with the useWorkspace hook resetting the store if it doesn't match the requested workspace, there's still a brief instant where this can trigger a fetch otherwise.